server/app/src/main/java/io/whitefox/api/deltasharing/DeltaMappers.java

```diff
@@ -26,8 +26,7 @@ public static io.whitefox.api.deltasharing.model.v1.generated.Schema schema2api(
   }
 
   public static ReadTableRequest api2ReadTableRequest(QueryRequest request) {
-    if (request.getEndingVersion() != null || request.getStartingVersion() != null)
-      throw new NotImplementedYet();
+    if (request.getEndingVersion() != null) throw new NotImplementedYet();
```

Comment:
in this code path we are ignoring `request.startingVersion` even if it's not null since it's not mapped into a field on ReadTableRequest. Then it's not used to actually read the data from Delta Lake. Can you dig a bit deeper and understand how we should make use of it?